### PR TITLE
feat: Improve `extract_struct_from_enum_variant` output

### PR DESCRIPTION
**Pull request by DropDemBits**
Improves the struct generated by `extract_struct_from_enum_variant`.

Summary of changes:

- Indent the generated struct and enum to the same indent level
- Preserve comments & attributes from the enum variant (something I missed when doing the same thing for the variant fields)
- Use enum's visibility for fields without any visibility, instead of filling it in with `pub`

### CODE REVIEW

File: crates/ide_assists/src/handlers/extract_struct_from_enum_variant.rs [hunk 1]
```diff
@@ -204,7 +206,34 @@ fn create_struct_def(
...
+                    return true;
+                }
+                WHITESPACE if select_next_ws => true,
+                _ => false,
+            };
+            select_next_ws = false;
+
+            accept
+        })
+    };
```
Comment (DropDemBits) [inline crates/ide_assists/src/handlers/extract_struct_from_enum_variant.rs]: ^ this

Modified from `remove_attrs_and_docs` in `AttrsOwnerEdit`, which is the reasoning behind the potential location.
I can open a separate issue + pr (or just a pr) for this later though.

File: crates/ide_assists/src/handlers/extract_struct_from_enum_variant.rs [hunk 2]
```diff
@@ -152,48 +162,81 @@ fn existing_definition(db: &RootDatabase, variant_name: &ast::Name, variant: &Va
...
+    field_list.reindent_to(IndentLevel::single());
+
     // FIXME: This uses all the generic params of the enum, but the variant might not use all of them.
-    let strukt =
-        make::struct_(enum_.visibility(), variant_name, enum_.generic_param_list(), field_list)
-            .clone_for_update();
+    let strukt = make::struct_(enum_vis, variant_name, enum_.generic_param_list(), field_list)
+        .clone_for_update();
+
+    // FIXME: Consider making this an actual function somewhere (like in `AttrsOwnerEdit`) after some deliberation
```
Comment (matklad) [inline crates/ide_assists/src/handlers/extract_struct_from_enum_variant.rs:212]: Yeah, very much this. 

I *think* we want to add this to `DocCommentsOwner` as

```rust
fn outer_attrs_and_docs(&self) -> Vec<SyntaxElement>
// alternative
fn outer_attrs_and_docs(&self) -> RangeInclusive<SyntaxElemet>
```

then `Edit` will contain logic for only actually deliting that stuff.

File: crates/ide_assists/src/handlers/extract_struct_from_enum_variant.rs [hunk 3]
```diff
@@ -104,10 +106,18 @@ pub(crate) fn extract_struct_from_enum_variant(
...
+            def.reindent_to(indent);
+
             let start_offset = &variant.parent_enum().syntax().clone();
-            ted::insert_raw(ted::Position::before(start_offset), def.syntax());
-            ted::insert_raw(ted::Position::before(start_offset), &make::tokens::blank_line());
+            ted::insert_all_raw(
+                ted::Position::before(start_offset),
+                vec![
+                    def.syntax().clone().into(),
+                    make::tokens::whitespace(&format!("\n\n{}", indent)).into(),
```
Comment (matklad) [inline crates/ide_assists/src/handlers/extract_struct_from_enum_variant.rs:118]: Ideally, you'd be able to `insert_all` here, and have the auto-ws magic to do the indentation for you. Might be an interesting case to look at after this PR is merged.

**Comment by matklad** [comment]
bors r+

Nice! I really like these kinds of PR -- by now, we have *a lot* of assists, but they are very rough along the edges, making them do the right thing improves the UX a low!

**Pull request merged**